api/v1alpha1/envoyproxy_types.go

```diff
@@ -86,6 +87,20 @@ type EnvoyProxySpec struct {
 	//
 	// +optional
 	Shutdown *ShutdownConfig `json:"shutdown,omitempty"`
+
+	// TLS is the TLS configuration for the Envoy proxy to use when connecting to
+	// backend services and external services, such as ExtAuth, ALS, OpenTelemetry, etc.
+	// +optional
+	TLS *EnvoyTLSConfig `json:"tls,omitempty"`
```

Comment:
should we call this ClientTLS ?

Reply:
@guydc mentioned that he wants to a[dd some server-side setting](https://github.com/envoyproxy/gateway/issues/2901)s later. If that's the case, then it would be better to use a more generalized name.